services/wallet/datastore.go

```diff
@@ -1069,6 +1089,58 @@ func (pg *Postgres) ConnectCustodialWallet(ctx context.Context, cl *CustodianLin
 	return nil
 }
 
+// InsertVerifiedWalletOutboxTx inserts a verifiedWalletOutbox for processing.
+func (pg *Postgres) InsertVerifiedWalletOutboxTx(ctx context.Context, tx *sqlx.Tx, walletID uuid.UUID, verifiedWallet bool) error {
+	_, err := tx.ExecContext(ctx, `insert into verified_wallet_outbox(payment_id, verified_wallet) 
+											values ($1, $2)`, walletID, verifiedWallet)
+	if err != nil {
+		return fmt.Errorf("error inserting values into vefified wallet outbox: %w", err)
+	}
+	return nil
+}
+
+// SendVerifiedWalletOutbox sends requests to reputation service.
+func (pg *Postgres) SendVerifiedWalletOutbox(ctx context.Context, client reputation.Client, retry backoff.RetryFunc) (bool, error) {
+	vw := struct {
+		ID             uuid.UUID `db:"id"`
+		PaymentID      uuid.UUID `db:"payment_id"`
+		VerifiedWallet bool      `db:"verified_wallet"`
+	}{}
+
+	_, tx, rollback, commit, err := datastore.GetTx(ctx, pg)
+	if err != nil {
+		return false, fmt.Errorf("error getting tx for verified wallet: %w", err)
+	}
+	defer rollback()
+
+	err = tx.Get(&vw, `select id, payment_id, verified_wallet from verified_wallet_outbox 
+                                   order by created_at asc for update skip locked limit 1`)
+	if err != nil {
+		return false, fmt.Errorf("error get verified wallet: %w", err)
+	}
+
+	upsertReputationSummaryOp := func() (interface{}, error) {
+		return nil, client.UpdateReputationSummary(ctx, vw.PaymentID.String(), vw.VerifiedWallet)
+	}
+
+	_, err = retry(ctx, upsertReputationSummaryOp, retryPolicy, canRetry(nonRetriableErrors))
```

Comment:
this will block each transaction?  we have many many linking attempts, maybe get a better picture on the scale from server logs to linking endpoing?

Reply:
Which txn are we talking about? This will not block the linking call from the client that is done in a separate txn which simply inserts into this table. This is running separately and a separate txn and will select one row in the order they were inserted and make the call. 

Also, in 2.5 we will only ever link once.